src/client/Client.js

```diff
@@ -351,6 +364,72 @@ class Client extends BaseClient {
     return messages;
   }
 
+  /**
+   * Removes all the stale data in the caches older than the provided lifetimes.
+   * @param {Lifetimes} lifetime Data that has been stale for longer than
+   * this will be removed.
+   * @returns {number} -1 if the lifetime is invalid and 1 on success
+   * @example
+   * // Remove all the stale cached user data older than 1800 seconds
+   * client.sweepCaches({ user: 1800 * 1000 }); // x 1000 to convert to seconds
+   */
+  sweepCaches(lifetime) {
+    if (typeof lifetime !== 'object') {
+      throw new TypeError('INVALID_TYPE', 'lifetime', 'Lifetimes');
+    }
+
+    const { user, channel, guild, emoji } = lifetime;
+    if (user !== undefined && (typeof user !== 'number' || isNaN(user))) {
+      throw new TypeError('INVALID_TYPE', 'lifetime.user', 'number or undefined');
+    }
+    if (channel !== undefined && (typeof channel !== 'number' || isNaN(channel))) {
+      throw new TypeError('INVALID_TYPE', 'lifetime.channel', 'number or undefined');
+    }
+    if (guild !== undefined && (typeof guild !== 'number' || isNaN(guild))) {
+      throw new TypeError('INVALID_TYPE', 'lifetime.guild', 'number or undefined');
+    }
+    if (emoji !== undefined && (typeof emoji !== 'number' || isNaN(emoji))) {
+      throw new TypeError('INVALID_TYPE', 'lifetime.emoji', 'number or undefined');
+    }
+
+    this.emit(Events.DEBUG, 'Cache sweep initiated, polling all the caches');
+
+    if (user !== undefined) {
+      if (user <= 0) {
+        this.emit(Events.DEBUG, "Didn't sweep user cache - lifetime is unlimited");
+      } else {
+        this.users.poll(user);
+      }
+    }
+
+    if (channel !== undefined) {
+      if (channel <= 0) {
+        this.emit(Events.DEBUG, "Didn't sweep channel cache - lifetime is unlimited");
+      } else {
+        this.channels.poll(channel);
+      }
+    }
+
+    if (guild !== undefined) {
+      if (guild <= 0) {
+        this.emit(Events.DEBUG, "Didn't sweep guild cache - lifetime is unlimited");
+      } else {
+        this.guilds.poll(guild);
+      }
+    }
+
+    if (emoji !== undefined) {
+      if (emoji <= 0) {
+        this.emit(Events.DEBUG, "Didn't sweep emoji cache - lifetime is unlimited");
+      } else {
+        this.emojis.poll(emoji);
+      }
+    }
+
+    this.emit(Events.DEBUG, `Swept all the stale data older than ${lifetime / 1000} seconds`);
+    return 1;
```

Comment:
why return 1 ?

Reply:
Whoops! I copied the `Client#sweepMessages` method and modified it for `Client#sweepCaches` but at some point forgot to remove the unnecessary return values, fixed it.